task/buildah/0.4/ARTIFACT_REUSE_FEATURE.md

```diff
@@ -0,0 +1,180 @@
+# Artifact Reuse Feature Implementation
+
+## Overview
+
+This document describes the implementation of an artifact reuse feature in the Konflux CI/CD system to avoid redundant builds, saving time, workspace quota, and cloud costs.
+
+## Problem Statement
+
+- **Redundant builds**: Identical source content produces identical artifacts regardless of git commit metadata
+- **Resource waste**: Time, compute resources, and storage costs for unnecessary rebuilds
+- **Use cases**: Re-running failed pipelines, reusing PR artifacts in push pipelines
+
+## Technical Design
+
+### Core Concept: Git Tree Hash as Content Fingerprint
+
+- **Tree hash**: `git show -s --format="%T" <commit>` provides content-based fingerprint
+- **Content identity**: Identical source trees produce identical tree hashes regardless of commit metadata
+- **Architecture independence**: Each architecture build examined individually for reuse
+
+### Build Configuration Fingerprint
+
+**Parameters compared for reuse** (excluding `IMAGE_EXPIRES_AFTER`):
```

Comment:
Let's rephrase this to indicate that it should take into account every parameter and that there will be some very specific exclusions. Trying to avoid the case where a new parameter is added but the re-use logic doesn't explicitly handle it - that would not be ideal.

Reply:
done